godot-core/src/builtin/packed_array.rs

```diff
@@ -258,6 +258,34 @@ macro_rules! impl_packed_array {
                 self.as_inner().sort();
             }
 
+            /// Extracts a slice of the entire array.
+            pub fn as_slice<'a>(&'a self) -> &'a [$Element] {
+                let len = self.len();
+
+                if len == 0 {
+                    return &[];
+                }
+
+                let data = self.ptr(0);
+                unsafe {
```

Comment:
Could you maybe write a 
```rs
// SAFETY: explanation here
```
section before the `unsafe` block, justifying why it's OK to use that here? (I'm not arguing it isn't, but it's good to have design decisions documented) 🙂